static/js/tools/stat_var/dataset_selector.tsx

```diff
@@ -0,0 +1,231 @@
+            dcid: props.sourceMap[filteredSuggestions[activeSuggestion]],
+            name: filteredSuggestions[activeSuggestion],
+          },
+        ]);
+      }
+    } else if (e.key === "ArrowUp") {
+      if (activeSuggestion === 0) {
+        return;
+      }
+      setActiveSuggestion(activeSuggestion - 1);
+    } else if (e.key === "ArrowDown") {
+      if (activeSuggestion - 1 === filteredSuggestions.length) {
+        return;
+      }
+      setActiveSuggestion(activeSuggestion + 1);
+    }
+  };
+
+  return (
+    <>
+      <Card className={`${PREFIX}-card`}>
+        <Container fluid={true} className={`${PREFIX}-container`}>
+          <div className={`${PREFIX}-main-selector`}>
+            <div className={`${PREFIX}-section`}>
+              <div className={`${PREFIX}-label`}>Show variables for</div>
+              <div className={`${PREFIX}-search`}>
+                <div className={`${PREFIX}-source-field`}>
+                  <input
+                    id={`${PREFIX}-ac`}
+                    className={`${PREFIX}-ac`}
+                    type="text"
+                    placeholder="Enter a source to filter by"
+                    onChange={handleOnChange}
+                    onKeyDown={handleOnKeyDown}
+                    value={userInput}
+                  />
+                  <i className={`material-icons ${PREFIX}-search-icon`}>
+                    search
+                  </i>
+                </div>
+                {showSuggestions && userInput && (
+                  <ul
+                    className={`${PREFIX}-suggestions`}
+                    style={{ width: `${suggestionsWidth || 0}px` }}
+                  >
+                    {filteredSuggestions.map((s, i) => {
+                      let className: string;
+                      if (i === activeSuggestion) {
+                        className = `${PREFIX}-suggestion-active`;
+                      }
+                      return (
+                        <li
+                          className={className}
+                          onClick={handleOnClick}
+                          key={s}
+                        >
+                          {s}
+                        </li>
+                      );
+                    })}
+                  </ul>
+                )}
+              </div>
+            </div>
+          </div>
+          <div className={`${PREFIX}-section`}>
+            <CustomInput
+              id={`${PREFIX}-custom-input`}
+              type="select"
+              onChange={(e) => {
+                const name = e.currentTarget.value
+                  ? e.currentTarget.innerText
+                  : userInput;
+                const dcid = e.currentTarget.value
+                  ? e.currentTarget.value
+                  : props.sourceMap[userInput];
+                props.filterStatVars([{ name, dcid }]);
+              }}
+            >
+              <option value="">Optionally select a dataset</option>
```

Comment:
are users supposed to do one of search for a source OR select a dataset?

Reply:
dataset is subset of source, so they can first search for source and then optionally further filter by dataset

Reply:
Do they have to have a source selected first in order to filter by dataset? Because the wording "optionally select a dataset" leads me to believe I can either search for a source or choose a dataset instead .. maybe "select a dataset (optional)"?

Reply:
oh okay I changed to this instead!